include/exadg/vector_tools/arbitrary_solution_transfer.h

```diff
@@ -0,0 +1,167 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_VECTOR_TOOLS_ARBITRARY_SOLUTION_TRANSFER_H
+#define INCLUDE_VECTOR_TOOLS_ARBITRARY_SOLUTION_TRANSFER_H
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/fe/mapping.h>
+#include <deal.II/matrix_free/fe_point_evaluation.h>
+#include <deal.II/numerics/vector_tools.h>
+
+namespace ExaDG
+{
+template<int dim>
+class ArbitrarySolutionTransfer
+{
+public:
+  /**
+   * Set source and target triangulations on which solution transfer is performed.
+   *
+   * @param[in] dof_handler_dst_in Target DofHandler.
+   * @param[in] mapping_dst_in Target Mapping.
+   * @param[in] dof_handler_src_in Source DofHandler.
+   * @param[in] mapping_src_in Source Mapping.
+   */
+  void
+  reinit(dealii::DoFHandler<dim> const & dof_handler_dst_in,
+         dealii::Mapping<dim> const &    mapping_dst_in,
+         dealii::DoFHandler<dim> const & dof_handler_src_in,
+         dealii::Mapping<dim> const &    mapping_src_in)
+  {
+    dof_handler_dst = &dof_handler_dst_in;
+    dof_handler_src = &dof_handler_src_in;
+
+    rpe.reinit(collect_mapped_support_points(dof_handler_dst_in, mapping_dst_in),
+               dof_handler_src_in.get_triangulation(),
+               mapping_src_in);
+  }
+
+  /**
+   * Transfer the solution between source and target triangulations, that were set with reinit.
+   * At support points which are not overlapping the source triangulation, corresponding dof entries
+   * are set to 0.
+   *
+   * @param[in] dst Target Dof Vector.
+   * @param[in] src Source Dof Vector.
+   */
+  template<int n_components, typename VectorType1, typename VectorType2>
+  void
+  transfer_solution(VectorType1 &                                               dst,
+                    VectorType2 const &                                         src,
+                    dealii::VectorTools::EvaluationFlags::EvaluationFlags const flags =
+                      dealii::VectorTools::EvaluationFlags::avg) const
+  {
+    src.update_ghost_values();
+    const auto values =
+      dealii::VectorTools::point_values<n_components>(rpe, *dof_handler_src, src, flags);
+    src.zero_out_ghost_values();
+
+    fill_dof_vector_with_values<n_components>(dst, *dof_handler_dst, values);
+  }
+
+private:
+  std::vector<dealii::Point<dim>>
+  collect_mapped_support_points(dealii::DoFHandler<dim> const & dof_handler,
+                                dealii::Mapping<dim> const &    mapping) const
+  {
+    std::vector<dealii::Point<dim>> global_mapped_support_points;
+    std::vector<dealii::Point<dim>> mapped_support_points;
```

Comment:
I think the term "nodes" would be more appropriate here? Then, one could distinguish between "global_nodes" and "elementwise/cellwise_nodes".

Reply:
What do you think about "support_points" and "cellwise_support_points"? I would like to reflect where points are evaluated in the name, and nodes is in my opinion not as clear.

Reply:
I can live with support points as well.